File: src/main/java/fi/quanfoxes/assembler/references/AddressReference.java
package fi.quanfoxes.assembler.references;

import fi.quanfoxes.assembler.*;

public class AddressReference extends Reference {
    private Number number;

    public AddressReference(Number number) {
        super(Size.DWORD);
        this.number = number;
    }

    @Override
    public String use(Size size) {
        return String.format("%s [%d]", size, number.longValue());
    }

    @Override
    public boolean isComplex() {
        return true;
    }

    public LocationType getType() {
        return LocationType.ADDRESS;
    }
}

File: Zigzag/Assembler/References/AddressReference.cs
public class AddressReference : Reference
{
	private object Value { get; set; }

	public AddressReference(object value) : base(Size.DWORD)
	{
		Value = value;
	}

	public override string Use(Size size)
	{
		return $"{size} [{(long)Value}]";
	}

	public override bool IsComplex()
	{
		return true;
	}

	public override LocationType GetType()
	{
		return LocationType.ADDRESS;
	}
}